File: chart/src/main/java/cn/jingzhuan/lib/chart/component/AxisX.java
package cn.jingzhuan.lib.chart.component;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Created by Donglua on 17/7/17.
 */

public class AxisX extends Axis {

    public final static int TOP = 101;
    public final static int BOTTOM = 102;
    public final static int TOP_INSIDE = 103;
    public final static int BOTTOM_INSIDE = 104;


    @Retention(RetentionPolicy.SOURCE)
    @IntDef({TOP, BOTTOM, TOP_INSIDE, BOTTOM_INSIDE})
    @interface AxisXPosition {}

    public AxisX(@AxisXPosition int axisPosition) {
        super(axisPosition);
    }



}

File: chart/src/main/java/cn/jingzhuan/lib/chart3/axis/AxisX.kt
package cn.jingzhuan.lib.chart3.axis

import androidx.annotation.IntDef

/**
 * @since 2023-09-06
 * X轴
 */
class AxisX(@AxisXPosition axisPosition: Int) : Axis(axisPosition) {

    @Retention(AnnotationRetention.SOURCE)
    @IntDef(TOP, BOTTOM, TOP_INSIDE, BOTTOM_INSIDE)
    internal annotation class AxisXPosition

    companion object {
        const val TOP = 101
        const val BOTTOM = 102
        const val TOP_INSIDE = 103
        const val BOTTOM_INSIDE = 104
    }
}
